equistore-core/include/equistore.hpp

```diff
@@ -1630,6 +1630,24 @@ class TensorBlock final {
         return array;
     }
 
+    /// Release the `eqs_block_t` pointer corresponding to this `TensorBlock`.
+    ///
+    /// The block pointer is **no longer** managed by the current `TensorBlock`,
+    /// and should manually be freed when no longer required.
+    eqs_block_t* release() {
+         if (is_view_) {
+            throw Error(
+                "can not call TensorBlock::release on this "
+                "block since it is a view inside a TensorMap"
+            );
+        }
+        auto ptr = block_;
+        block_ = nullptr;
+        is_view_ = false;
+        return ptr;
+    }
+
+    friend class TensorMap;
```

Comment:
is this from debugging? I dont see any reason why this is needed for this PR

Reply:
TensorMap uses the private function to move blocks around